beacon-chain/core/blocks/block_operations.go

```diff
@@ -621,18 +621,29 @@ func ProcessValidatorDeposits(
 		if err != nil {
 			return nil, fmt.Errorf("could not process deposit into beacon state: %v", err)
 		}
+
+		beaconState.DepositIndex++
 	}
 	return beaconState, nil
 }
 
 func verifyDeposit(beaconState *pb.BeaconState, deposit *pb.Deposit) error {
+	// Deposits must be processed in order
+	if deposit.MerkleTreeIndex != beaconState.DepositIndex {
+		return fmt.Errorf(
+			"expected deposit merkle branch index to match beacon state deposit index, wanted: %d, received: %d",
```

Comment:
Merkle branch index or merkle tree index?

Reply:
Good catch, thank you. Changed to merkle tree index.